### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by antonioeatgoat**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?**
Dependencies update

**What is the current behavior?**
Current dependencies doesn't support PHP 8

**What is the new behavior?**
The new dependencies support PHP 8

**Does this PR introduce a breaking change?**
Yes. As described [here](https://github.com/WordPress/WordPress-Coding-Standards/wiki/Upgrade-Guide-to-WordPressCS-3.0.0-for-ruleset-maintainers#ruleset-and-phpcs-ignore-annotations), the name of some rules of WorrdPress Coding Standards are changed and so any direct referral (for instance `phpcs:ignore RULE_NAME) would not work anymore.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,11 +29,10 @@
...
     "minimum-stability": "stable",
     "require": {
-        "php": ">=7",
+        "php": ">=7.4",
         "squizlabs/php_codesniffer": "^3.6.0",
-        "dealerdirect/phpcodesniffer-composer-installer": "~0.7.0",
-        "wp-coding-standards/wpcs": "^2.3",
-        "automattic/vipwpcs": "^2.2",
+        "dealerdirect/phpcodesniffer-composer-installer": "~1.0.0",
+        "automattic/vipwpcs": "dev-3.0/updates-for-wpcs-3.0",
```
Comment (TangRufus) [inline composer.json:35]: `3.0/updates-for-wpcs-3.0` has been merged 1 minute ago

https://github.com/Automattic/VIP-Coding-Standards/pull/779

**Review by TangRufus — CHANGES_REQUESTED** [review]

**Review by gmazzap — APPROVED** [review]

**Comment by gmazzap** [comment]
@antonioeatgoat Already updated the requirement for `automattic/vipwpcs` as requested by @TangRufus 

- Optimized dependencies to reduce conflicts and for PHP 7.4+
- Fixed tests failing because usage of changed PHPCS API
- Made use of Inpsyde reusable workflow for tests

Now I think this is mergiable.

**Comment by gmazzap** [comment]
@antonioeatgoat @TangRufus We'll need to release is a nww major, so I created the `version/2` branch and changed the base branch to that.

**Comment by gmazzap** [comment]
This has been merged into #62 so I'm closing this

**Pull request closed without merging**